### PR TITLE
Sync packages for WP 6.1 RC 1 (with Fluid Typography)

### PR DESCRIPTION
**Pull request by ockham**
_Alternative to #3434. It's also possible to merge #3434 first, and then merge this PR, which contains all the changes from #3434, **plus** the Fluid Typography fixes (see https://github.com/WordPress/gutenberg/issues/44758 and https://github.com/WordPress/wordpress-develop/pull/3431)._

Includes the latest round of Gutenberg bugfix PRs that were approved for inclusion during [yesterday’s triage session](https://wordpress.slack.com/archives/C02QB2JS7/p1665408036271379). The following Gutenberg PRs were cherry-picked:

- https://github.com/WordPress/gutenberg/pull/44640
- https://github.com/WordPress/gutenberg/pull/44652
- https://github.com/WordPress/gutenberg/pull/44584
- https://github.com/WordPress/gutenberg/pull/43181
- https://github.com/WordPress/gutenberg/pull/44650
- https://github.com/WordPress/gutenberg/pull/44731
- https://github.com/WordPress/gutenberg/pull/44660
- https://github.com/WordPress/gutenberg/pull/44718
- https://github.com/WordPress/gutenberg/pull/44721
- https://github.com/WordPress/gutenberg/pull/44463
- https://github.com/WordPress/gutenberg/pull/44415
- https://github.com/WordPress/gutenberg/pull/44676
- https://github.com/WordPress/gutenberg/pull/44150
- https://github.com/WordPress/gutenberg/pull/44098
- https://github.com/WordPress/gutenberg/pull/44637

Furthermore, this includes the Fluid Typography fixes listed in https://github.com/WordPress/gutenberg/issues/44758.

Needed for today's WP 6.1 RC 1 release.

Trac ticket: https://core.trac.wordpress.org/ticket/56467

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/blocks/navigation-link.php [hunk 1]
```diff
@@ -98,7 +98,14 @@ function block_core_navigation_link_build_css_font_sizes( $context ) {
 		$font_sizes['css_classes'][] = sprintf( 'has-%s-font-size', $context['fontSize'] );
 	} elseif ( $has_custom_font_size ) {
 		// Add the custom font size inline style.
-		$font_sizes['inline_styles'] = sprintf( 'font-size: %s;', $context['style']['typography']['fontSize'] );
+		$font_sizes['inline_styles'] = sprintf(
+			'font-size: %s;',
+			gutenberg_get_typography_font_size_value(
```
Comment (dream-encode) [inline src/wp-includes/blocks/navigation-link.php:103]:
```suggestion
			wp_get_typography_font_size_value(
```

File: src/wp-includes/blocks/navigation-submenu.php [hunk 2]
```diff
@@ -98,7 +98,14 @@ function block_core_navigation_submenu_build_css_font_sizes( $context ) {
 		$font_sizes['css_classes'][] = sprintf( 'has-%s-font-size', $context['fontSize'] );
 	} elseif ( $has_custom_font_size ) {
 		// Add the custom font size inline style.
-		$font_sizes['inline_styles'] = sprintf( 'font-size: %s;', $context['style']['typography']['fontSize'] );
+		$font_sizes['inline_styles'] = sprintf(
+			'font-size: %s;',
+			gutenberg_get_typography_font_size_value(
```
Comment (dream-encode) [inline src/wp-includes/blocks/navigation-submenu.php:103]:
```suggestion
			wp_get_typography_font_size_value(
```

File: src/wp-includes/blocks/page-list.php [hunk 3]
```diff
@@ -119,7 +119,14 @@ function block_core_page_list_build_css_font_sizes( $context ) {
 		$font_sizes['css_classes'][] = sprintf( 'has-%s-font-size', $context['fontSize'] );
 	} elseif ( $has_custom_font_size ) {
 		// Add the custom font size inline style.
-		$font_sizes['inline_styles'] = sprintf( 'font-size: %s;', $context['style']['typography']['fontSize'] );
+		$font_sizes['inline_styles'] = sprintf(
+			'font-size: %s;',
+			gutenberg_get_typography_font_size_value(
```
Comment (dream-encode) [inline src/wp-includes/blocks/page-list.php:124]:
```suggestion
			wp_get_typography_font_size_value(
```

File: src/wp-includes/blocks/search.php [hunk 4]
```diff
@@ -442,31 +442,39 @@ function get_typography_styles_for_block_core_search( $attributes ) {
 
 	// Add typography styles.
 	if ( ! empty( $attributes['style']['typography']['fontSize'] ) ) {
-		$typography_styles[] = sprintf( 'font-size: %s;', esc_attr( $attributes['style']['typography']['fontSize'] ) );
+		$typography_styles[] = sprintf(
+			'font-size: %s;',
+			gutenberg_get_typography_font_size_value(
```
Comment (dream-encode) [inline src/wp-includes/blocks/search.php:447]:
```suggestion
			wp_get_typography_font_size_value(
```

File: src/wp-includes/blocks/search.php [hunk 5]
```diff
@@ -442,31 +442,39 @@ function get_typography_styles_for_block_core_search( $attributes ) {
...
 	}
 
 	if ( ! empty( $attributes['style']['typography']['lineHeight'] ) ) {
-		$typography_styles[] = sprintf( 'line-height: %s;', esc_attr( $attributes['style']['typography']['lineHeight'] ) );
+		$typography_styles[] = sprintf( 'line-height: %s;', $attributes['style']['typography']['lineHeight'] );
 	}
 
 	if ( ! empty( $attributes['style']['typography']['textTransform'] ) ) {
-		$typography_styles[] = sprintf( 'text-transform: %s;', esc_attr( $attributes['style']['typography']['textTransform'] ) );
+		$typography_styles[] = sprintf( 'text-transform: %s;', $attributes['style']['typography']['textTransform'] );
```
Comment (dream-encode) [inline src/wp-includes/blocks/search.php:477]: I feel like this code could be simplified to one loop over all elements in `$attributes['style']['typography']`. 

Inside the loop, we could abstract out the building of the inline styles to another function.  Also, if there are more style props added, this code would only need to be changed if there's some sort of unique transform needed on the value, e.g. `fontSize` key above on line 447 passes its value through `wp_get_typography_font_size_value`.

 Example:

```
$allowed_keys = array(
    'fontFamily',
    'letterSpacing',
    'fontWeight',
    'fontStyle',
    'lineHeight',
    'textTransform',   
);

if ( isset( $attributes['style']['typography'] ) ) {
    foreach ( $attributes['style']['typography'] as $key => $value ) {
        if ( in_array( $key, $allowed_keys, true ) && ! empty( $attributes['style']['typography'][ $key ] ) ) {
            $typography_styles[] = sprintf( 
                '%1$s: %2$s;',
                _wp_to_kebab_case( $key ),
                $attributes['style']['typography'][ $key ]
            );
        }
    }
}

Comment (dream-encode) [reply src/wp-includes/blocks/search.php:477]: Edited to add an allowlist of known keys, to avoid parsing bad ones.

**Review by dream-encode — APPROVED** [review]
I think this code is mostly OK as-is, aside from updating the function prefix from `gutenberg` to `wp`.

My comments on the inline styles aren't a blocker for 6.1 RC 1, but I think this code should be refactored.  There's actually a lot of this pattern in(at least) the search block code, i.e. check if value for specific key in array has a value, and, if so, a similar task to several other keys.

**Comment by ockham** [comment]
> I think this code is mostly OK as-is, aside from updating the function prefix from `gutenberg` to `wp`.

Good catch! Since this is dynamic block PHP, we'll have to do that in Gutenberg, as that code is sync'ed from the `@wordpress/block-library` package during Core's build process.

**Comment by dream-encode** [comment]
Merged into core in https://core.trac.wordpress.org/changeset/54483.

**Comment by ockham** [comment]
> Merged into core in https://core.trac.wordpress.org/changeset/54483.

I believe @dream-encode merged the version without the Fluid Typography changes (https://github.com/WordPress/wordpress-develop/pull/3434). Reopening 😅

**Comment by ockham** [comment]
> > I think this code is mostly OK as-is, aside from updating the function prefix from `gutenberg` to `wp`.
> 
> Good catch! Since this is dynamic block PHP, we'll have to do that in Gutenberg, as that code is sync'ed from the `@wordpress/block-library` package during Core's build process.

https://github.com/WordPress/gutenberg/pull/44876

**Comment by ockham** [comment]
> > Merged into core in https://core.trac.wordpress.org/changeset/54483.
> 
> I believe @dream-encode merged the version without the Fluid Typography changes (#3434). Reopening 😅

Ah, looks like it _was_ actually this one that was merged! Apologies for the confusion. Closing again; https://github.com/WordPress/wordpress-develop/pull/3439 is the follow-up to prevent the `gutenberg_` prefix from being re-written during build.

**Pull request closed without merging**